From: Ubuntu Developers <ubuntu-devel-discuss@lists.ubuntu.com>
Date: Mon, 16 Mar 2026 10:31:00 +0800
Subject: bash53-006

BASH PATCH REPORT
			     =================

Bash-Release:	5.3
Patch-ID:	bash53-006

Bug-Reported-by:	Duncan Roe <duncan_roe@optusnet.com.au>
Bug-Reference-ID:	<aL0p/3BQeCLc6krz@dimstar.local.net>
Bug-Reference-URL:	https://lists.gnu.org/archive/html/bug-bash/2025-09/msg00053.html

Bug-Description:

When `globasciiranges' is enabled, glob patterns with ranges in bracket
expressions can produce incorrect matches for character ranges whose
start and end are non-ascii characters.
---
 lib/glob/smatch.c | 2 +-
 patchlevel.h      | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/glob/smatch.c b/lib/glob/smatch.c
index e42b8fc..af3856a 100644
--- a/lib/glob/smatch.c
+++ b/lib/glob/smatch.c
@@ -390,7 +390,7 @@ charcmp_wc (wint_t c1, wint_t c2, int forcecoll)
   if (c1 == c2)
     return 0;
 
-  if (forcecoll == 0 && glob_asciirange && c1 <= UCHAR_MAX && c2 <= UCHAR_MAX)
+  if (forcecoll == 0 && glob_asciirange)
     return ((int)(c1 - c2));
 
   s1[0] = c1;
diff --git a/patchlevel.h b/patchlevel.h
index 7caf230..b8e99ec 100644
--- a/patchlevel.h
+++ b/patchlevel.h
@@ -25,6 +25,6 @@
    regexp `^#define[ 	]*PATCHLEVEL', since that's what support/mkversion.sh
    looks for to find the patch level (for the sccs version string). */
 
-#define PATCHLEVEL 5
+#define PATCHLEVEL 6
 
 #endif /* _PATCHLEVEL_H_ */
